front-end/public/background.js

```diff
@@ -12,46 +10,80 @@ try {
     title: "Add to Workspace",
     contexts: ["selection"]
   };
-
+  
   // Adding and removing the items
   //--------------------------------------------------------------
-
+  
   // Make sure any other instance of our menus are removed first
   chrome.contextMenus.removeAll();
-
+  
   chrome.contextMenus.create(parentContextMenuItem);
-
-  // Add all the workspaces as children
-  fetch(`${serverAddr}/workspaces`)
-    .then((r) => r.text())
-    .then((result) => {
-      // Result now contains the response text, do what you want...
-      const workspaces = JSON.parse(result);
-      workspaces.map((workspace) =>
-        chrome.contextMenus.create({
-          id: workspace._id,
-          title: workspace.name,
-          contexts: ["selection"],
-          parentId: "parent",
-        })
-      );
-    });
+  
+  var currUserId;
+  var workspaceIds = [];
+  var isLogged = false;
+  chrome.runtime.onMessageExternal.addListener(
+    function(request) {
+      // if there is no account logged in locally
+      if (request.messageFromWeb.auth0Id === "") {
+        isLogged = false;
+        workspaceIds = [];
+        currUserId = "";
+        chrome.contextMenus.removeAll();
+        chrome.contextMenus.create(parentContextMenuItem);
+      }
+      // else, there is an account/user logged in
+      else {
+        fetch(`${helper.serverAddr}/accounts`)
+          .then((r) => r.text())
+          .then((account_result) => {
+            // Result now contains the response text, do what you want...
+            const accounts = JSON.parse(account_result);
+            accounts.map((account) => {
+              // if user is currently logged in
+              if (request.messageFromWeb.auth0Id === account.auth0Id) {
+                isLogged = true;
+                if (currUserId === account.auth0Id) {
+                  helper.createWorkspaceContextMenus(workspaceIds, account);
+                }
+                // if current user is different from currently logged user
+                else {
+                  workspaceIds = [];
+                  currUserId = account.auth0Id;
+                  // Add all the workspaces as children
+                  fetch(`${helper.serverAddr}/workspaces`)
```

Comment:
There's also an endpoint `/workspaces/byOwner/:ownerId` that you could use to avoid filtering. Would have to add some stuff later on to also get workspaces you don't own but are a collaborator of, so filtering also works if you dont feel like doin that LOL